src/javascript/javascript-import-scanner.ts

```diff
@@ -254,32 +236,30 @@ export class JavaScriptImportScanner implements JavaScriptScanner {
     if (isWindows() && documentPath.startsWith('/')) {
       documentPath = documentPath.substring(1);
     }
-    let resolvedSpecifier;
-    try {
-      resolvedSpecifier = resolve.sync(specifier, {basedir: documentPath});
-    } catch (e) {
-      warnings.push(new Warning({
-        message: 'Cannot resolve module specifier',
-        sourceRange: document.sourceRangeForNode(node)!,
-        severity: Severity.WARNING,
-        code: 'cant-resolve-module-specifier',
-        parsedDocument: document,
-      }));
-      return undefined;
-    }
-
-    let relativeSpecifierUrl =
-        relative(dirname(documentPath), resolvedSpecifier) as FileRelativeUrl;
 
-    if (isWindows()) {
-      // normalize path separators to URL format
-      relativeSpecifierUrl =
-          relativeSpecifierUrl.replace(/\\/g, '/') as FileRelativeUrl;
+    if (this.moduleResolution === 'node') {
+      try {
+        // TODO (justinfagnani): we need to pass packageName and componentDir,
+        // or maybe the UrlResolver, to get correct relative URLs for
+        // named imports from the root package to its dependencies
+        return resolve(specifier, documentPath);
+      } catch (e) {
+        // If the specifier was a name, we'll emit a warning below.
```

Comment:
The warning emitted below though is too specific, it's just for when there's no module resolution algorithm.